daprdocs/content/en/operations/components/pluggable-components/pluggable-components-discovery.md

```diff
@@ -0,0 +1,212 @@
+---
+type: docs
+title: "Plugging in"
+linkTitle: "Plugging in"
+weight: 4500
+description: "How-To: Making Dapr discover a pluggable component"
+---
+
+## Service Discovery Process
+
+<img src="/images/grpc-components.png" width=400>
+
+[gRPC-based](https://grpc.io/) Dapr components are typically run as containers or processes that communicate with the Dapr main process via [Unix Domain Sockets][uds]. They are automatically discovered and registered in runtime by Dapr using the following steps:
+
+1. The Component listens to an [Unix Domain Socket][uds] placed on the shared volume.
+2. The Dapr runtime lists all [Unix Domain Socket][uds] in the shared volume.
+3. The Dapr runtime connects with the socket and uses gRPC reflection to discover all services that such component implements.
+
+A single component can implement multiple [building blocks](http://localhost:1313/concepts/building-blocks-concept/) at once.
+
+Dapr's built-in components come ready to be used out of the box: you just need to have provide Dapr with their YAML configuration and you are ready to go. That is not the case with gRPC-based Components, which require a few setup steps before they can be used with Dapr. Namely:
+
+1. gGRPC-based Components need to be started and ready to take requests _before_ Dapr itself is started;
+2. The [Unix Domain Socket][uds] file used used for the Pluggable Component communication need to be made accessible to both Dapr and gRPC Component and, as expected,
```

Comment:
I think the original list had 3 steps. Now the second one is not finished and is finishing on `and, as expected`


```suggestion
2. The [Unix Domain Socket][uds] file used used for the Pluggable Component communication need to be made accessible to both Dapr and gRPC Component.
```